src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodeParagraphDescriptions.kt

```diff
@@ -0,0 +1,132 @@
+  YOI_8("Escapes or absconds from a young offender institution or from legal custody"),
+  YOI_9("Failing to comply with conditions of release under rule 5 which relates to temporary release"),
+  YOI_10("Is found with any substance in his urine which demonstrates that a controlled drug, pharmacy medication, prescription only medicine, psychoactive substance or specified substance has, whether in prison or while on temporary release under rule 5, been administered to him by himself or by another person (but subject to Rule 56)"),
+  YOI_12("Knowingly consumes any alcoholic beverage other than that provided to them pursuant to a written order under rule 21 (alcohol and tobacco) (1)"),
+  YOI_16("Takes improperly any article belonging to another person or to a young offender institution"),
+  YOI_17("Intentionally or recklessly sets fire to any part of a young offender institution or any other property, whether or not his own"),
+  YOI_27("Receives any controlled drug, or, without the consent of an officer, any other article, during the course of a visit (not being an interview such as is mentioned in rule 16)"),
+  YOI_1_ADULT_1("Commits any assault"),
+  YOI_3_ADULT_2("Detains any person against their will"),
+  YOI_5_ADULT_4("Fights with any person"),
+  YOI_6_ADULT_5("Intentionally endangers the health or personal safety of others or, by their conduct, is reckless whether such health or personal safety is endangered"),
+  YOI_11_ADULT_10("Is intoxicated as a consequence of knowingly consuming any alcoholic beverage"),
+  YOI_13_ADULT_12("Has in his possession either (a) any unauthorised article, or (b) a greater quantity of any article than he is authorised to have"),
+  YOI_14_ADULT_13("Sells or delivers to any person any unauthorised article"),
+  YOI_15_ADULT_14("Sells or, without permission, delivers to any person any article which they are allowed to have only for their own use"),
+  YOI_19_ADULT_24A("Displays, attaches or draws on any part of a prison, or on any other property, threatening, abusive or insulting racist words, drawings, symbols or other material"),
+  YOI_20_ADULT_18("Absents himself from any place he is required to be or is present at any place where they are not authorised to be"),
+  YOI_21_ADULT_19("Is disrespectful to any officer, or any person (other than a prisoner) who is at the prison for the purpose of working there, or any person visiting a prison"),
+  YOI_22_YOI_23_ADULT_20("Uses threatening, abusive or insulting words or behaviour"),
+  YOI_24_ADULT_21("Intentionally fails to work properly or, being required to work, refuses to do so"),
+  YOI_25_ADULT_22("Disobeys any lawful order"),
+  YOI_26_ADULT_23("Disobeys or fails to comply with any rule or regulation applying to them"),
+  DEFAULT("")
+}
+
+class OffenceCodeParagraphs {
+
+  private val lookup = mapOf(
+    "55:2" to "YOI_1_ADULT_1",
```

Comment:
We could make this a `Map<String, Descriptions>` for example `"55:2" to Descriptions.YOI_1_ADULT_1` which would simplify `getParagraphDescription`?

Reply:
Yeah good idea.  It was a string as i generated it that way, but makes sense now the test confirmed the mappings are correct.